components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -116,6 +116,117 @@ std::string URLMethodToRequestType(ledger::type::UrlMethod method) {
   }
 }
 
+bool ResetOnFileTaskRunner(
+    base::WeakPtr<base::File> diagnostic_log,
+    const std::vector<std::string>& paths) {
```

Comment:
We do a lot of string/FilePath conversions here and the other log file functions - can we just send FilePaths over to the file sequence to avoid the conversions?

Reply:
I was trying to avoid a situation where I pass a `FilePath` member variable and the service goes away before the callback is called. I felt like passing a `std::string` and then reconstructing it in the callback was safer and avoided that possibility.

Reply:
`FilePath` values are copyable, so I think the values will be copied in to the bound function by `BindOnce`?

Reply:
OK, will fix.